src/lib/Quiz.jsx

```diff
@@ -26,28 +78,41 @@ const Quiz = function ({
   }, []);
 
   useEffect(() => {
+    // declaring a common variale for all operation
+    let newQuestions = quiz.questions;
+
+    // shuffling questions  if shuffle is true
     if (shuffle) {
-      setQuestions(shuffleQuestions(quiz.questions));
-    } else {
-      quiz.questions.length = nrOfQuestions;
-      setQuestions(quiz.questions);
+      newQuestions = shuffleQuestions(newQuestions);
     }
 
-    setQuestions(questions.map((question, index) => ({
+    // shuffling answers if shuffleAnswer is true
```

Comment:
same as above.

Reply:
send a PR,
i cant figure out how to fix this,
i had to ctrl+z (undo) many thing, and then tried again.
i hope now it will work 😔😔😔